lib/reverse_sentence.rb

```diff
@@ -1,6 +1,54 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+# Time complexity: O(n^2), there is a nested loop in #reverse_word, which runs n operations dependent on the lengths of find_spaces and my_sentence
+# Space complexity: O(n), a new element is pushed to the find_spaces array for each space contained in my_sentence
+
+def reverse_by_character(my_sentence)
```

Comment:
Outstanding submission, I would say the space complexity is O(1) if you assume only a few spaces relative to the size of the array but you are correct that does add memory usage.